develop/tutorials/articles/260-forms/01-form-field-types/02-adding-form-field-type-settings.markdown

```diff
@@ -227,15 +214,9 @@ The interface `DDMFormFieldTemplateContextContributor` only has one single metho
         }
     )
 
-Then, override the method `getParameters` and get the new configurations settings, `placeholder` and `mask`:
+Then override `getParameters` to get the new configurations settings,
+`placeholder` and `mask`:
 
-    @Component(
```

Comment:
I think we should keep the component code or just leave the overridden method on this code snippet, what do you think?

Reply:
What if we move the class declaration up so it reads like this:

> First create its OSGI component annotation and the class declaration:
```
    @Component(
        immediate = true,
        property = "ddm.form.field.type.name=time",
        service = {
            DDMFormFieldTemplateContextContributor.class,
            TimeDDMFormFieldTemplateContextContributor.class
        }
    )
    public class TimeDDMFormFieldTemplateContextContributor
        implements DDMFormFieldTemplateContextContributor {

```
> Then override `getParameters` to get the new configurations settings, `placeholder` and `mask`:
```

        @Override
        public Map<String, Object> getParameters(
            DDMFormField ddmFormField,
            DDMFormFieldRenderingContext ddmFormFieldRenderingContext) {

            Map<String, Object> parameters = new HashMap<>();

            parameters.put(
                "placeholder", (String)ddmFormField.getProperty("placeholder"));
            parameters.put("mask", (String)ddmFormField.getProperty("mask"));

            return parameters;
        }

    }

```

Reply:
Nice!